src/Terrain/TerrainType.java

```diff
@@ -52,134 +54,142 @@ private TerrainType(int attributeFlags, int defense, Color mainColor, String nam
   public Boolean healsLand() { return 0 != (mAttributes & HEALS_LAND); }
   public Boolean healsSea() { return 0 != (mAttributes & HEALS_SEA); }
   public Boolean healsAir() { return 0 != (mAttributes & HEALS_AIR); }
+  public TerrainType getBaseTerrain() { return (null == mBase)? SEA : mBase; }
   
   ////////////////////////////////////////////////////////////////////////////////////////////
   //// Publicly-accessible terrain Flyweight instances.
-  private static final int AIRPORT_FLAGS = LAND | CAPTURABLE | PROFITABLE | PROVIDES_COVER | HEALS_AIR;
-  private static final int AIRPORT_DEFENSE = 3;
-  private static final Color AIRPORT_COLOR = new Color(125, 125, 125); // TODO: define unique color for each building type?
-  private static final String AIRPORT_NAME = "AIRPORT";
-  public static final TerrainType AIRPORT = new TerrainType( AIRPORT_FLAGS, AIRPORT_DEFENSE, AIRPORT_COLOR, AIRPORT_NAME);
+  private static final int SEA_FLAGS = WATER;
```

Comment:
Why reorder all of these?

Reply:
I needed to do some reordering because e.g. SEA has to be constructed before anything else can use it as a parameter in their own constructors.
I was originally planning to try to impose some sort of logical ordering, but realized I didn't know exactly what kind I wanted :P